nestkernel/connector_base.h

```diff
@@ -434,14 +462,14 @@ class Connector : public vector_like< ConnectionT >
   send( Event& e, thread t, const std::vector< ConnectorModel* >& cm )
   {
     synindex syn_id = C_[ 0 ].get_syn_id();
+    typename ConnectionT::CommonPropertiesType const& cp =
+      static_cast< GenericConnectorModel< ConnectionT >* >( cm[ syn_id ] )
+        ->get_common_properties();
     for ( size_t i = 0; i < K; i++ )
     {
       e.set_port( i );
-      C_[ i ].send( e,
-        t,
-        ConnectorBase::get_t_lastspike(),
-        static_cast< GenericConnectorModel< ConnectionT >* >( cm[ syn_id ] )
-          ->get_common_properties() );
+      C_[ i ].send( e, t, ConnectorBase::get_t_lastspike(), cp );
+      ConnectorBase::send_weight_event( cp, e, t );
```

Comment:
yes, you're right. they do not contribute to MPI communication, but adds an additional `if` for every spike event that is delivered. turns out it's not a performance issue (see below).